GraphsV2/ChordGraph/index.js

```diff
@@ -0,0 +1,320 @@
+    if (data && data.destination) {
+        const { accessor, label } = (
+            (tooltip && tooltip.length === 1)
+                ? { accessor: columnAccessor(tooltip[0]), label: tooltip[0].label }
+                : { accessor: (d) => d.value, label: undefined }
+        );
+        return chordTooltipContent(data, accessor, label)
+    } else {
+        return <div>Hover over a chord to see flow details.</div>;
+    }
+}
+
+const chordTooltipContent = (data, accessor, label) => {
+    const {
+        source,
+        destination,
+        sourceValue,
+        destinationValue,
+        isBidirectionalTooltip
+    } = data;
+
+    return (
+        <React.Fragment>
+            {renderTooltipContent(destination, source, sourceValue, accessor, label)}
+            {isBidirectionalTooltip && renderTooltipContent(source, destination, destinationValue, accessor, label)}
+        </React.Fragment>
+    );
+}
+
+const renderTooltipContent = (from, to, value, accessor, label) => (
+    <div>
+        <strong>{`${from} to ${to}:`}</strong>
+        <span> {accessor({ value })}</span>
+        {label ? <span> {label}</span> : null}
+    </div>
+);
+
+const weight = (d, chordWeightColumn) => d[chordWeightColumn];
+const destination = (d, chordDestinationColumn) => d[chordDestinationColumn];
+const source = (d, chordSourceColumn) => d[chordSourceColumn]; 
+
+const generateMatrix = ({
+    data, 
+    chordWeightColumn, 
+    chordSourceColumn,
+    chordDestinationColumn
+}) => {
+    if (isEmpty(data)) {
+        return []
+    }
+
+    let indices = {},
+        matrix = [],
+        names = [],
+        matrixData = [],
+        n = 0, i, j;
+
+    const recordIndex = (name) =>{
+        if (!(name in indices)) {
+            indices[name] = n++;
+            names.push(name);
+        }
+    }
+
+    data.forEach(d => {
+        recordIndex(source(d, chordSourceColumn));
+        recordIndex(destination(d, chordDestinationColumn));
+    });
+
+    for (i = 0; i < n; i++) {
+        matrix.push([]);
+        matrixData.push([]);
+        for (j = 0; j < n; j++) {
+            matrix[i].push(0);
+            matrixData[i].push([]);
+        }
+    }
+
+    data.forEach(d => {
+        i = indices[source(d, chordSourceColumn)];
+        j = indices[destination(d, chordDestinationColumn)];
+
+        if (chordWeightColumn) {
+            matrix[j][i] += weight(d, chordWeightColumn);
+        } else {
+
+            // Handle the case where no weight column was specified
+            // by making the chord weight fixed on both sides.
+            matrix[j][i] = matrix[i][j] = 1;
+        }
+        matrixData[j][i].push(d);
+    });
+
+    matrix.names = names;
+    matrix.data = matrixData;
+
+    return matrix;
+}
+
+const getFilterData = (data, chordSourceColumn, chordDestinationColumn) => {
+    return data.filter(d => d[chordSourceColumn] && d[chordDestinationColumn]);
+}
+
+let filterChordData = [];
```

Comment:
just a question: why did you choose to use these instead of saving the data in the component's state, this way the change in the data will cause a re-render and that is what we want isn't it?